### PR TITLE
Ensure header link buttons work when pressing space.

### PR DESCRIPTION
**Pull request by nickcolley**
Some assistive technologies, like screen readers, suggest to press 'space' when interacting with a link with a role of 'button'.
We need to ensure that we replicate this functionality that exists on a button element.
In the future we should consider using the 'button' element instead.

I decided against changing the link to a button to avoid a breaking change, and @notlee has let me know that the anchor is used in the 'core' experience to link to the bottom of the page.

I'm wondering if it's better to put this space handling next to the calls to `Toggle`, but went with this top level handler so there's only one event listener... Welcome your thoughts there.

Closes https://github.com/Financial-Times/o-header/issues/380

### CODE REVIEW

File: test/helpers/events.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export function dispatch (target, type, eventProperties) {
+	const event = new Event(type, { bubbles: true });
+	for (const property in eventProperties) {
+		event[property] = eventProperties[property];
+	}
```
Comment (JakeChampion) [inline test/helpers/events.js:5]:
```suggestion
	}
```
need to remove the trailing space to appease the linting rules

File: test/helpers/events.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export function dispatch (target, type, eventProperties) {
+	const event = new Event(type, { bubbles: true });
+	for (const property in eventProperties) {
```
Comment (JakeChampion) [inline test/helpers/events.js:3]: This fails on the linting rules currently:
```
/test/helpers/events.js:3:2 Error - The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype. (guard-for-in)
```

Comment (nickcolley) [reply test/helpers/events.js:3]: Makes sense

**Comment by JakeChampion** [comment]
Thanks for this work @nickcolley it is really good. I like your idea about how to go about deprecating the link markup as well :-)

**Comment by notlee** [comment]
Your suggestions to use `button` in the future are sensible 👍 I think an anchor is used currently for the core experience, to link to the drawer content when JavaScript is unavailable; customer products put drawer content below the footer on ft.com presumably to avoid the layout jumping around as o-header JS is initalised. We'd need to keep that in mind in a future major release

Screenshot 2020-06-05 at 12 12 08

**Comment by nickcolley** [comment]
> I think an anchor is used currently for the core experience, to link to the drawer content when JavaScript is unavailable

This feels like a valid reason to be using an anchor here, I guess if we did move towards using a button we'd want to replace the anchor element with a button with JavaScript...

**Comment by nickcolley** [comment]
Is there anything I need to do for a changelog entry?

**Comment by JakeChampion** [comment]
We don't _yet_ have changelogs for each repo, it is something we need to discuss in the future for sure :-)

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v8.3.0 :tada:

**Pull request merged**